### PR TITLE
Proposal changes

### PR DESCRIPTION
**Pull request by hidiego**
#### What kind of PR is this?:

/kind fix

#### What this PR does / why we need it:
Renaming and adding other useful information to proposals

### CODE REVIEW

File: src/masternodes/rpc_proposals.cpp [hunk 1]
```diff
@@ -554,8 +554,8 @@ UniValue getgovproposal(const JSONRPCRequest& request)
     ret.pushKV("proposalId", propId.GetHex());
     ret.pushKV("title", prop->title);
     ret.pushKV("context", prop->context);
-    if (!prop->contexthash.empty())
-        ret.pushKV("contexthash", prop->contexthash);
+    if (!prop->contextHash.empty())
+        ret.pushKV("contexthash", prop->contextHash);
```
Comment (prasannavl) [inline src/masternodes/rpc_proposals.cpp:558]: Change. Please do a Ctrl F and verify all changes.

File: src/masternodes/proposals.cpp [hunk 2]
```diff
@@ -88,6 +88,7 @@ std::optional<CPropObject> CPropsView::GetProp(const CPropId& propId)
         if (auto cycle = ReadBy<ByStatus, uint8_t>(key)) {
             prop->cycle = *cycle;
             prop->status = status;
+            prop->cycleEndHeight = prop->creationHeight + (prop->votingPeriod - prop->creationHeight % prop->votingPeriod) + prop->votingPeriod;
```
Comment (Jouzo) [inline src/masternodes/proposals.cpp:91]:
```suggestion
            prop->cycleEndHeight = prop->creationHeight + (prop->votingPeriod - prop->creationHeight % prop->votingPeriod) + prop->votingPeriod * *cycle;
```

File: src/masternodes/rpc_proposals.cpp [hunk 3]
```diff
@@ -577,14 +582,11 @@ UniValue getgovproposal(const JSONRPCRequest& request)
...
-        ret.pushKV("approval", strprintf("%d.%02d of %d.%02d%%", votes / 100, votes % 100, majorityThreshold / 100, majorityThreshold % 100));
-    } else {
-        ret.pushKV("validity", strprintf("%d.%02d of %d.%02d%%", allVotes / 100, allVotes % 100, quorum / 100, quorum % 100));
-    }
+    ret.pushKV("votes", strprintf("%d.%02d of %d.%02d%%", votes / 100, votes % 100, majorityThreshold / 100, majorityThreshold % 100));
+    ret.pushKV("votingPercent", strprintf("%d.%02d of %d.%02d%%", allVotes / 100, allVotes % 100, quorum / 100, quorum % 100));
 
     auto target = Params().GetConsensus().pos.nTargetSpacing;
-    auto blocks = prop->finalHeight - targetHeight;
+    auto blocks = prop->proposalEndHeight - targetHeight;
```
Comment (prasannavl) [inline src/masternodes/rpc_proposals.cpp:589]:
```suggestion
    auto blocks = prop->proposalEndHeight - (view.GetLastHeight() + 1);
```

[the remark this answers is not in the document]
Comment (prasannavl) [reply src/masternodes/rpc_proposals.cpp:589]:
```suggestion
    auto blocks = prop->proposalEndHeight - (view.GetLastHeight() + 1);
```

[the remark this answers is not in the document]
Comment (hidiego) [reply src/masternodes/rpc_proposals.cpp:589]: Agree, ends returns the approximate time which can be calculated with proposalEndHeight/cycleEndHeight not so relevant. Also ends format is not standard for API users to parse easily.

**Pull request merged**